src/main/java/com/venafi/vcert/sdk/connectors/cloud/CloudConnectorUtils.java

```diff
@@ -174,28 +174,38 @@ private static void addCitToApp(CertificateIssuingTemplate cit, Application appl
             application.certificateIssuingTemplateAliasIdMap( citAliasIdMap );
         }
 
+		boolean updateApplication = false;
+
         //if the App doesn't contain the relation to the cit
         if ( !citAliasIdMap.containsKey(cit.name()) ) {
             //adding the reference to the cit
             citAliasIdMap.put(cit.name(), cit.id());
+            updateApplication = true;
+        }
 
-            //getting the appId because it will be used to invoke the API to update the related Application
-            String appId = application.id();
+		// Updating the owners list of the Application
+		List<Application.OwnerIdsAndType> ownersList =  CloudConnectorUtils.resolveUsersToCloudOwners(usersList, apiKey, cloud);
```

Comment:
I think that the merge only should happen if the owners in the ownersList is different to the owners of the application

Reply:
Iterating the lists to find if they are different or not is expensive and unnecessary when they are going to be iterated again in order to be merged/filtered. Instead, doing both things in one single iteration is faster.

Reply:
Mmmm... in fact I was wrong in my previous comment, the Application's Owners should be always overwritten with the Owners list gotten from the PolicySpecification's Users